### DISCUSSION TITLE
"Error getting node" after restart of cluster

### DISCUSSION THREAD

**Discussion by sven-borkert** [okd-project/okd#1398] [Q&A] [answered]
Hi,

I have shut down my testing cluster (4.11.0-0.okd-2022-11-05-030711) over night and tried to start it again now, beginning with the 3 master nodes, then the 2 workers. But nothing seems to start at all. All the master nodes are complaining:

Nov 09 14:06:38 master0.ocp4.borkert.net hyperkube[2743]: E1109 14:06:38.740402    2743 kubelet.go:2447] "Error getting node" err="node \"master0.ocp4.borkert.net\" not found"

They all say they cannot find themselves, and "oc get nodes" shows them all as NotReady. I waited an hour, but they don't start again. Can I not just shut down the cluster and start it again later? What do I have to read to understand what to do now to get it back working?

Regardsm
Sven

**Comment by JaimeMagiera** [comment, marked as answer]
Check if you have pending certificates. 

`oc get csr`

**Reply by sven-borkert** [reply]
Oh yes, there suddenly are a lot of them again. I approved them all and the cluster went back to normal. Sorry for the stupid question. I had confirmed all the CSRs after installation, I did not expect them to re-appear after a reboot. Seems there is still a lot to learn for me here. Thank you.

**Comment by bdlink** [comment]
In the docs under backup there is information on safe shutdown and restart which mentions approving CSRs as necessary.  It is necessarily after a long shutdown.

Sent from my iPhone

On Nov 9, 2022, at 7:16 AM, Sven Borkert ***@***.***> wrote:

﻿
CAUTION: This email originated from outside of BCIT. Do not click links or open attachments unless you recognize the sender and know the content is safe.


Oh yes, there suddenly are a lot of them again. I approved them all and the cluster went back to normal. Sorry for the stupid question. I had confirmed all the CSRs after installation, I did not expect them to re-appear after a reboot. Seems there is still a lot to learn for me here. Thank you.
